doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,394 @@
+>>> from sympy.parsing.sympy_parser import convert_equals_signs, standard_transformations
+>>> expr = "x ** 2 = y"
+>>> parsed = parse_expr(expr, transformations=standard_transformations + (convert_equals_signs,))
+>>> print(parsed)
+Eq(x**2, y)
+>>> solutions = solveset(parsed, x)
+>>> print(solutions)
+{-sqrt(y), sqrt(y)}
+```
+
+If you already have the equation in `Eq` form, you can parse that string:
+
+```py
+>>> from sympy import parse_expr, solveset
+>>> from sympy.abc import x
+>>> expr = "Eq(x**2, y)"
+>>> parsed = parse_expr(expr)
+>>> print(parsed)
+Eq(x**2, y)
+>>> solutions = solveset(parsed, x)
+>>> print(solutions)
+{-sqrt(y), sqrt(y)}
+```
+
+### {func}`~.solveset` can return infinitely many solutions when {func}`~.solve` cannot
+
+{func}`~.solveset` 
+[can return infinitely many solutions](https://docs.sympy.org/dev/modules/solvers/solveset.html?highlight=solveset#why-solveset) 
+and express them in standard mathematical notation, 
+for example $\sin(x) = 0$ for $x = n * \pi$ for every integer value of $n$:
+
+```py
+>>> from sympy import pprint, sin, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(sin(x), x)
+>>> pprint(solution)
+{2*n*pi | n in Integers} U {2*n*pi + pi | n in Integers}
+```
+
+However, {func}`~.solve` will return only a finite number of solutions:
+
+```py
+>>> from sympy import sin, solve
+>>> from sympy.abc import x
+>>> solution = solve(sin(x), x)
+>>> print(solution)
+[0, pi]
+```
+
+SymPy tries to return just enough solutions so that all (infinitely many) solutions can generated 
+from the returned solutions by adding integer multiples of the periodicity of the equation, here $2\pi$.
+
+### Restricting the domain of solutions using {func}`~.solveset`
+
+By default, SymPy will return solutions in the complex domain, which also includes purely real and 
+imaginary values. Here, the first two solutions are real, and the last two are imaginary:
+
+```py
+>>> from sympy import S, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(x**4 - 256, x)
+>>> print(solution)
+{-4, 4, -4*I, 4*I}
+```
+
+If you want to restrict returned solutions to real numbers, you can specify the 
+[domain to solve in](https://docs.sympy.org/dev/modules/solvers/solveset.html?highlight=solveset#what-is-this-domain-argument-about) 
+as `S.Reals`:
+
+```py
+>>> from sympy import S, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(x**4 - 256, x, domain=S.Reals)
+>>> print(solution)
+{-4, 4}
+```
+
+## Not all equations can be solved
+
+### Equations with no solution
+
+Some equations have no solution, in which case SymPy may return an empty set. For example, the equation 
+$x - 7 = x + 2$ reduces to $-7 = 2$, which has no solution because no value of $x$ will make it true:
+
+```py
+>>> from sympy import solve, Eq
+>>> from sympy.abc import x
+>>> eqn = Eq(x - 7, x + 2)
+>>> solve(eqn, x)
+[]
+```
+
+So if SymPy returns an empty set, you may want to check whether there is a mistake in the equation.
```

Comment:
This is an empty list rather than an empty set. It's important to be clear here because `solveset` returns the mathematical empty set.